### PR TITLE
Project One Time Access Claims

### PR DESCRIPTION
**Pull request by bbengfort**
### Scope of changes

Implements an endpoint in Quarterdeck to issue a user one-time access to an Ensign project.

Fixes SC-13958

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

See checklist below.

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [ ]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?
- [ ] Is the expiration of the claim long enough/short enough?
- [ ] Does it make sense to only copy over the topic related permissions?
- [ ] Are there any claims missing in the response?
- [ ] Is it ok to reuse the login reply struct?

### CODE REVIEW

File: pkg/quarterdeck/orgs.go [hunk 1]
```diff
@@ -113,3 +119,80 @@ func (s *Server) ProjectCreate(c *gin.Context) {
...
+		},
+		OrgID:       claims.OrgID,
+		ProjectID:   project.ProjectID.String(),
+		Permissions: make([]string, 0, 4),
+	}
+
+	// Add only the user permissions related to topics to these claims -- whatever
+	// access to topics the user has, so to will the one time access claims.
+	for _, permission := range claims.Permissions {
+		if strings.HasPrefix(permission, "topics") {
```
Comment (pdeziel) [inline pkg/quarterdeck/orgs.go:184]: Should we define the permission prefixes in code to protect ourselves from future permissions changes?

File: pkg/quarterdeck/orgs_test.go [hunk 2]
```diff
@@ -104,3 +106,69 @@ func (s *quarterdeckTestSuite) TestProjectCreate() {
...
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// Requesting one-time access to a project requires authentication
+	req := &api.Project{ProjectID: ulid.MustParse("01GQ7P8DNR9MR64RJR9D64FFNT")}
+	_, err := s.client.ProjectAccess(ctx, req)
+	s.CheckError(err, http.StatusUnauthorized, "this endpoint requires authentication")
+
+	// Requesting one-time access to a project an Project requires the topics:read permission
```
Comment (pdeziel) [inline pkg/quarterdeck/orgs_test.go:122]:
```suggestion
	// Requesting one-time access to a project requires the topics:read permission
```

**Review by pdeziel — APPROVED** [review]
Makes sense to me, I think reusing the `LoginReply` is fine and the claim expiration also makes sense. I was trying to think if this could be abused by a user to achieve more permissions than they have but I see we're only copying over the ones that already exist in the claims, so looks good! I had a few small comments but it shouldn't block the merge.

File: pkg/quarterdeck/orgs.go (hunk 1 above)
Comment (bbengfort) [reply pkg/quarterdeck/orgs.go:184]: Good idea

**Pull request merged**